pkg/driver/credential.go

```diff
@@ -138,7 +145,16 @@ func (c *credentialProvider) provideFromPod(ctx context.Context, req *csi.NodePu
 	}, nil
 }
 
-func (c *credentialProvider) findPodServiceAccountRole(ctx context.Context, req *csi.NodePublishVolumeRequest) (string, error) {
+func (c *CredentialProvider) writeToken(volumeID string, token *Token) error {
+	// TODO: Should we make the write atomic by writing to a temporary path and renaming afterwards?
```

Comment:
There's a [function in `driver.go`](https://github.com/awslabs/mountpoint-s3-csi-driver/blob/fa790569624fb917f68887274ce6fb79cbc969a9/pkg/driver/driver.go#L135-L180) that does this atomically. We can reuse it.

Reply:
Seems like that function works with existing files, I'll modify or create a new function for writing in-memory content